dask_geopandas/clip.py

```diff
@@ -0,0 +1,38 @@
+import numpy as np
+import geopandas
+
+from dask.highlevelgraph import HighLevelGraph
+from dask.utils import derived_from
+from dask.base import tokenize
+
+from .core import GeoDataFrame
+
+
+@derived_from(geopandas.tools)
+def clip(gdf, mask, keep_geom_type=False):
+    if gdf.spatial_partitions is None:
+        return gdf.map_partitions(
+            lambda partition: geopandas.clip(
+                gdf=partition, mask=mask, keep_geom_type=keep_geom_type
+            ),
+            token="clip",
+            meta=gdf._meta,
+        )
+
+    new_spatial_partitions = geopandas.clip(
+        gdf=gdf.spatial_partitions.to_frame("geometry"),
+        mask=mask,
+    )
+    intersecting_partitions = np.asarray(new_spatial_partitions.index)
```

Comment:
I assume this will only work if you have a default integer index. Now, this if for the spatial_partitions (not the actual gdf), so there we can actually expect/enforce this. 
Not for this PR, but we should probably add a check when creating the GeoDataFrame that the `spatial_partitions` has a default index.

Reply:
-> opened https://github.com/geopandas/dask-geopandas/issues/66 about this